pkg/proxy/healthcheck/healthcheck_test.go

```diff
@@ -411,3 +415,52 @@ func testHealthzHandler(server httpServer, status int, t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestServerWithSelectiveListeningAddress(t *testing.T) {
+	listener := newFakeListener()
+	httpFactory := newFakeHTTPServerFactory()
+
+	/*
+		limiting addresses to loop back. We don't want any cleverness here around getting IP for
+		machine nor testing ipv6 || ipv4. using loop back guarantees the test will work on any machine
+	*/
```

Comment:
we use //-style comments pretty much everywhere

Reply:
ack